@angular-cli 1.1.0 rc.1 not support baseUrl import ?

<!--
IF YOU DON'T FILL OUT THE FOLLOWING INFORMATION YOUR ISSUE MIGHT BE CLOSED WITHOUT INVESTIGATING
-->
### Bug Report or Feature Request (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [x] feature request
```

### Versions.
<!--
Output from: `ng --version`.
If nothing, output from: `node --version` and `npm --version`.
  Windows (7/8/10). Linux (incl. distribution). macOS (El Capitan? Sierra?)
-->
@angular/cli: 1.1.0-rc.1
node: 6.10.2
os: darwin x64
@angular/animations: 4.1.3
@angular/common: 4.1.3
@angular/compiler: 4.1.3
@angular/core: 4.1.3
@angular/forms: 4.1.3
@angular/http: 4.1.3
@angular/platform-browser: 4.1.3
@angular/platform-browser-dynamic: 4.1.3
@angular/router: 4.1.3
@angular/cli: 1.1.0-rc.1
@angular/compiler-cli: 4.1.3
@angular/language-service: 4.1.3

### Repro steps.
<!--
Simple steps to reproduce this bug.
Please include: commands run, packages added, related code changes.
A link to a sample repo would help too.
-->
ng new angular-new-cli
yarn install
new until.ts in app/
this is until.ts content export const TEST = 'test';
in app.component.ts import { TEST } from 'app/until'; 

### The log given by the failure.
<!-- Normally this include a stack trace and some more information. -->
ERROR in ./src/app/app.component.ts
Module not found: Error: Can't resolve 'app/until' in '/Users/feng/ng/angular-new-cli/src/app'
 @ ./src/app/app.component.ts 8:0-33
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts

### Desired functionality.
<!--
What would like to see implemented?
What is the usecase?
-->
@angular-cli 1.1.0 rc.0 -> 1.1.0 rc.1 not support baseUrl import ?
if have multi-level directory
how to do more elegant
### Mention any other details that might be useful.
<!-- Please include a link to the repo if this is related to an OSS project. -->

--- a/package.json
+++ b/package.json
@@ -50,6 +50,7 @@
     "diff": "^3.1.0",
     "ember-cli-normalize-entity-name": "^1.0.0",
     "ember-cli-string-utils": "^1.0.0",
+    "enhanced-resolve": "^3.1.0",
     "exports-loader": "^0.6.3",
     "extract-text-webpack-plugin": "^2.1.0",
     "file-loader": "^0.10.0",

--- a/packages/@ngtools/webpack/package.json
+++ b/packages/@ngtools/webpack/package.json
@@ -25,6 +25,7 @@
     "npm": ">= 3.0.0"
   },
   "dependencies": {
+    "enhanced-resolve": "^3.1.0",
     "loader-utils": "^1.0.2",
     "magic-string": "^0.19.0",
     "source-map": "^0.5.6"

--- a/packages/@ngtools/webpack/src/paths-plugin.ts
+++ b/packages/@ngtools/webpack/src/paths-plugin.ts
@@ -1,25 +1,46 @@
 import * as path from 'path';
 import * as ts from 'typescript';
-import {Callback, Tapable, NormalModuleFactory, NormalModuleFactoryRequest} from './webpack';
+import {Request, ResolverPlugin, Callback, Tapable} from './webpack';
+
+
+const ModulesInRootPlugin: new (a: string, b: string, c: string) => ResolverPlugin
+  = require('enhanced-resolve/lib/ModulesInRootPlugin');
+
+interface CreateInnerCallback {
+  (callback: Callback<any>,
+   options: Callback<any>,
+   message?: string,
+   messageOptional?: string): Callback<any>;
+}
+
+const createInnerCallback: CreateInnerCallback
+  = require('enhanced-resolve/lib/createInnerCallback');
+const getInnerRequest: (resolver: ResolverPlugin, request: Request) => string
+  = require('enhanced-resolve/lib/getInnerRequest');
+
 
 function escapeRegExp(str: string): string {
   return str.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, '\\$&');
 }
 
+
 export interface PathsPluginOptions {
-  nmf: NormalModuleFactory;
   tsConfigPath: string;
   compilerOptions?: ts.CompilerOptions;
   compilerHost?: ts.CompilerHost;
 }
 
 export class PathsPlugin implements Tapable {
-  private _nmf: NormalModuleFactory;
   private _tsConfigPath: string;
   private _compilerOptions: ts.CompilerOptions;
   private _host: ts.CompilerHost;
+
+  source: string;
+  target: string;
+
+  private mappings: any;
+
   private _absoluteBaseUrl: string;
-  private _mappings: any[] = [];
 
   private static _loadOptionsFromTsConfig(tsConfigPath: string, host?: ts.CompilerHost):
       ts.CompilerOptions {
@@ -55,13 +76,15 @@ export class PathsPlugin implements Tapable {
       this._host = ts.createCompilerHost(this._compilerOptions, false);
     }
 
+    this.source = 'described-resolve';
+    this.target = 'resolve';
+
     this._absoluteBaseUrl = path.resolve(
       path.dirname(this._tsConfigPath),
       this._compilerOptions.baseUrl || '.'
     );
 
-    this._nmf = options.nmf;
-
+    this.mappings = [];
     let paths = this._compilerOptions.paths || {};
     Object.keys(paths).forEach(alias => {
       let onlyModule = alias.indexOf('*') === -1;
@@ -76,7 +99,7 @@ export class PathsPlugin implements Tapable {
           aliasPattern = new RegExp(`^${withStarCapturing}`);
         }
 
-        this._mappings.push({
+        this.mappings.push({
           onlyModule,
           alias,
           aliasPattern,
@@ -86,36 +109,66 @@ export class PathsPlugin implements Tapable {
     });
   }
 
-  apply(): void {
-    this._nmf.plugin('before-resolve', (request: NormalModuleFactoryRequest,
-        callback: Callback<any>) => {
-      for (let mapping of this._mappings) {
-        const match = request.request.match(mapping.aliasPattern);
-        if (!match) { continue; }
+  apply(resolver: ResolverPlugin): void {
+    let baseUrl = this._compilerOptions.baseUrl || '.';
 
-        let newRequestStr = mapping.target;
-        if (!mapping.onlyModule) {
-          newRequestStr = newRequestStr.replace('*', match[1]);
-        }
+    if (baseUrl) {
+      resolver.apply(new ModulesInRootPlugin('module', this._absoluteBaseUrl, 'resolve'));
+    }
 
-        const moduleResolver: ts.ResolvedModuleWithFailedLookupLocations =
-            ts.nodeModuleNameResolver(
-              newRequestStr,
-              this._absoluteBaseUrl,
-              this._compilerOptions,
-              this._host
-            );
-        const moduleFilePath = moduleResolver.resolvedModule ?
-            moduleResolver.resolvedModule.resolvedFileName : '';
-
-        if (moduleFilePath) {
-          return callback(null, Object.assign({}, request, {
-            request: moduleFilePath.includes('.d.ts') ? newRequestStr : moduleFilePath
-          }));
-        }
-      }
-
-      return callback(null, request);
+    this.mappings.forEach((mapping: any) => {
+      resolver.plugin(this.source, this.createPlugin(resolver, mapping));
     });
+  }
+
+  resolve(resolver: ResolverPlugin, mapping: any, request: any, callback: Callback<any>): any {
+    let innerRequest = getInnerRequest(resolver, request);
+    if (!innerRequest) {
+      return callback();
+    }
+
+    let match = innerRequest.match(mapping.aliasPattern);
+    if (!match) {
+      return callback();
+    }
+
+    let newRequestStr = mapping.target;
+    if (!mapping.onlyModule) {
+      newRequestStr = newRequestStr.replace('*', match[1]);
+    }
+    if (newRequestStr[0] === '.') {
+      newRequestStr = path.resolve(this._absoluteBaseUrl, newRequestStr);
+    }
+
+    let newRequest = Object.assign({}, request, {
+      request: newRequestStr
+    }) as Request;
+
+    return resolver.doResolve(
+      this.target,
+      newRequest,
+      `aliased with mapping '${innerRequest}': '${mapping.alias}' to '${newRequestStr}'`,
+      createInnerCallback(
+        function(err, result) {
+          if (arguments.length > 0) {
+            return callback(err, result);
+          }
+
+          // don't allow other aliasing or raw request
+          callback(null, null);
+        },
+        callback
+      )
+    );
+  }
+
+  createPlugin(resolver: ResolverPlugin, mapping: any): any {
+    return (request: any, callback: Callback<any>) => {
+      try {
+        this.resolve(resolver, mapping, request, callback);
+      } catch (err) {
+        callback(err);
+      }
+    };
   }
 }

--- a/packages/@ngtools/webpack/src/plugin.ts
+++ b/packages/@ngtools/webpack/src/plugin.ts
@@ -341,11 +341,7 @@ export class AotPlugin implements Tapable {
           cb();
         }
       });
-    });
-
-    compiler.plugin('normal-module-factory', (nmf: any) => {
       compiler.resolvers.normal.apply(new PathsPlugin({
-        nmf,
         tsConfigPath: this._tsConfigPath,
         compilerOptions: this._compilerOptions,
         compilerHost: this._compilerHost

--- a/packages/@ngtools/webpack/src/webpack.ts
+++ b/packages/@ngtools/webpack/src/webpack.ts
@@ -39,18 +39,6 @@ export interface NormalModule {
   resource: string;
 }
 
-export interface NormalModuleFactory {
-  plugin(
-    event: string,
-    callback: (data: NormalModuleFactoryRequest, callback: Callback<any>) => void
-  ): any;
-}
-
-export interface NormalModuleFactoryRequest {
-  request: string;
-  contextInfo: { issuer: string };
-}
-
 export interface LoaderContext {
   _module: NormalModule;
 

--- a/tests/e2e/tests/build/ts-paths.ts
+++ b/tests/e2e/tests/build/ts-paths.ts
@@ -13,10 +13,6 @@ export default function() {
       ],
       '@shared/*': [
         'app/shared/*'
-      ],
-      '*': [
-        '*',
-        'app/shared/*'
       ]
     };
   })
@@ -29,14 +25,12 @@ export default function() {
     import { meaning } from 'app/shared/meaning';
     import { meaning as meaning2 } from '@shared';
     import { meaning as meaning3 } from '@shared/meaning';
-    import { meaning as meaning4 } from 'meaning';
 
     // need to use imports otherwise they are ignored and
     // no error is outputted, even if baseUrl/paths don't work
     console.log(meaning)
     console.log(meaning2)
     console.log(meaning3)
-    console.log(meaning4)
   `))
   .then(() => ng('build'));
 }

